### PR TITLE
Small custom UI fixes

### PR DESCRIPTION
**Pull request by GuillaumeRx**
## **Description**

Small UI adjustments made to Snaps custom UI components. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27101?quickstart=1)

## **Manual testing steps**

1. Go to test snap
2. Try an example snap that renders custom UI

### **Before**

image

### **After**

image

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/snaps/snap-ui-form/snap-ui-form.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ export const SnapUIForm: FunctionComponent<SnapUIFormProps> = ({
       id={name}
       display={Display.Flex}
       flexDirection={FlexDirection.Column}
-      gap={2}
+      gap={4}
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-form/snap-ui-form.tsx:36]: Bumps the gap between the elements to 16px

File: ui/components/app/snaps/snap-ui-input/index.scss [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
-    &:has(.snap-ui-renderer__image) {
-      display: flex;
-    }
+  & .mm-text-field > .snap-ui-renderer__panel {
+    gap: 8px;
+  }
+
+  & .snap-ui-button {
+    display: flex;
+    padding-inline: 0px;
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-input/index.scss:8]: Fixes alignment and gap for elements wrapped by buttons in input accessories

File: ui/components/app/snaps/snap-ui-renderer/components/section.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ export const section: UIComponentFactory<SectionElement> = ({
       ...props,
       className: 'snap-ui-renderer__section',
       padding: 4,
+      gap: 2,
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-renderer/components/section.ts:24]: Requested by @eriknson - Adds a gap of 8px between elements in Section

File: ui/components/app/snaps/snap-ui-button/index.scss [hunk 4]
```diff
@@ -1,6 +1,8 @@
-.snap-ui-button {
+.snap-ui-renderer__button {
   background: none;
   text-align: center;
+  display: flex;
+  padding-inline: 0;
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-button/index.scss:5]: Fix issue with icons inside buttons not being centered

File: ui/components/app/snaps/snap-ui-renderer/components/container.ts [hunk 5]
```diff
@@ -82,7 +82,7 @@ export const container: UIComponentFactory<BoxElement> = ({
       className: 'snap-ui-renderer__container',
       style: {
         overflowY: 'auto',
-        paddingBottom: useFooter ? '74px' : 'initial',
+        paddingBottom: useFooter ? '80px' : 'initial',
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-renderer/components/container.ts:85]: Fix footer being too small for the buttons

File: ui/components/app/snaps/snap-ui-renderer/components/footer.ts [hunk 6]
```diff
@@ -22,7 +22,7 @@ export const DEFAULT_FOOTER = {
     backgroundColor: BackgroundColor.backgroundDefault,
     style: {
       boxShadow: 'var(--shadow-size-md) var(--color-shadow-default)',
-      height: '74px',
+      height: '80px',
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-renderer/components/footer.ts:25]: Fix footer being too small for the buttons

File: ui/components/app/snaps/snap-ui-selector/snap-ui-selector.tsx [hunk 7]
```diff
@@ -103,7 +107,10 @@ export const SnapUISelector: React.FunctionComponent<SnapUISelectorProps> = ({
     }
   }, [initialValue]);
 
-  const handleModalOpen = () => setIsModalOpen(true);
+  const handleModalOpen = (event: ReactMouseEvent<HTMLElement>) => {
+    event.preventDefault();
+    setIsModalOpen(true);
+  };
```
Comment (GuillaumeRx) [inline ui/components/app/snaps/snap-ui-selector/snap-ui-selector.tsx:113]: prevent sending the form submission event when clicking the selector

**Review by david0xd — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**